go/pkg/pwapi/api_admin-challenge-instance-add.go

```diff
@@ -0,0 +1,56 @@
+package pwapi
+
+import (
+	"context"
+
+	"pathwar.land/v2/go/pkg/errcode"
+	"pathwar.land/v2/go/pkg/pwdb"
+)
+
+func (svc *service) AdminChallengeInstanceAdd(ctx context.Context, in *AdminChallengeInstanceAdd_Input) (*AdminChallengeInstanceAdd_Output, error) {
+	if !isAdminContext(ctx) {
+		return nil, errcode.ErrRestrictedArea
+	}
+
+	if in == nil {
+		return nil, errcode.ErrMissingInput
+	}
+
+	var agent pwdb.Agent
+	err := svc.db.
+		Where(pwdb.SeasonChallenge{ID: in.ChallengeInstance.AgentID}).
+		Preload("Agent").
+		First(&agent).
+		Error
+	if err != nil {
+		return nil, errcode.ErrChallengeInstanceAdd.Wrap(err)
+	}
+
+	var flavor pwdb.ChallengeFlavor
+	err = svc.db.
+		Where(pwdb.SeasonChallenge{ID: in.ChallengeInstance.FlavorID}).
+		Preload("ChallengeFlavor").
+		First(&agent).
+		Error
+	if err != nil {
+		return nil, errcode.ErrChallengeInstanceAdd.Wrap(err)
+	}
+
+	challengeInstance := pwdb.ChallengeInstance{
+		Status:   in.ChallengeInstance.Status,
+		Agent:    &agent,
+		AgentID:  in.ChallengeInstance.AgentID,
+		Flavor:   &flavor,
+		FlavorID: in.ChallengeInstance.FlavorID,
```

Comment:
```suggestion
		Status:   in.ChallengeInstance.Status,
		AgentID:  in.ChallengeInstance.AgentID,
		FlavorID: in.ChallengeInstance.FlavorID,
```

use only the ID here